src/graph/planner/plan/PlanNode.cpp

```diff
@@ -330,6 +330,17 @@ void PlanNode::readVariable(Variable* varPtr) {
   qctx_->symTable()->readBy(varPtr->name, this);
 }
 
+bool PlanNode::isConcurrencySafeInputVar() const {
+  if (inputVars_.size() != 1) {
+    return true;
+  }
+  auto inputVar = inputVars_.front();
+  if (inputVar->readBy.size() < 2) {
+    return true;
+  }
+  return false;
```

Comment:
`return inputVar->readBy.size() < 2`

Reply:
ACK.

Reply:
Already delete this interface and replace by Executor::movable.